### PR TITLE
Remove JSONParseException and remove unused Reindex constants.

### PR DESCRIPTION
**Pull request by coreation**
### Changes

* Used JsonException where the deprecated JSONParseException was used. 
* Removed usages of the latter and removed unused/deprecated constants in Reindex.php.

### CODE REVIEW

File: src/Reindex.php [hunk 1]
```diff
@@ -31,17 +29,11 @@ class Reindex extends Param
...
-     * @deprecated since version 7.2.0, use a boolean as parameter instead.
-     */
-    public const WAIT_FOR_COMPLETION_FALSE = 'false';
     public const WAIT_FOR_ACTIVE_SHARDS = 'wait_for_active_shards';
     public const TIMEOUT = 'timeout';
     public const SCROLL = 'scroll';
     public const REQUESTS_PER_SECOND = 'requests_per_second';
     public const PIPELINE = 'pipeline';
-    public const SLICES = 'slices';
-    public const SLICES_AUTO = 'auto';
```
Comment (thePanz) [inline src/Reindex.php]: We should leave those two consts, as they can be defined in the Reindex request:
https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html

Comment (coreation) [reply src/Reindex.php]: @thePanz thanks, I was not aware of them being defined in ElasticSearch. So these constants are just there to reflect the API of ElasticSearch, even though they're not used in the library itself, correct?

Comment (coreation) [reply src/Reindex.php]: @thePanz I've undone the removal.

Comment (thePanz) [reply src/Reindex.php]: I guess the `setSlices(int $slices)` method is missing, which would then use those constants
Please leave them, I will provide a PR to use them later :)

Comment (coreation) [reply src/Reindex.php]: Ok, I've put them back, if that's everything, I'll leave it up to you to close the conversation.

**Review by thePanz — APPROVED** [review]

File: src/Reindex.php (hunk 1 above)
Comment (thePanz) [reply src/Reindex.php]: LGTM, would you please fix the CS warnings?
Then I can merge it
\cc @ruflin

Comment (coreation) [reply src/Reindex.php]: Got ya, strange though, if I run the same CS fixer command locally, all looks well. I've pushed some fixes based on the CI output.

**Review by thePanz — APPROVED** [review]

**Comment by thePanz** [comment]
Thanks @coreation , merged!

**Comment by coreation** [comment]
Thanks! I've opened a PR as per @ruflin 's request for the elasticsearch-php client. I think I can contribute some time to help out, but there's a bunch of information I'd need from more experienced contributors as the refactor isn't that straightforward. The PR can be viewed [here](https://github.com/ruflin/Elastica/pull/2168) @thePanz . Let's do this!

**Comment by ruflin** [comment]
One nit on this PR: Changelog is missing ;-) @coreation Would be great to have a follow up PR for it so we don't forget.

**Comment by coreation** [comment]
@ruflin no problem, I made a new PR [here](https://github.com/ruflin/Elastica/pull/2169). I fixed a small typo in a doc block as well. I wasn't sure where to add my changes, but I think the "Removed" section sounded appropriate ^^

**Pull request merged**